generic/Storage.c

```diff
@@ -39,6 +39,27 @@ static int torch_Storage_(new)(lua_State *L)
       lua_pop(L, 1);
     }
   }
+  else if(lua_type(L, index) == LUA_TUSERDATA)
+  {
+    THStorage *src = luaT_checkudata(L, index, torch_Storage);
+    if (src->flag & TH_STORAGE_VIEW) {
+      // don't allow chained views
+      src = src->view;
```

Comment:
This code works, no need to throw an error: if you have 3 storages, A, B and C, with A=>B, and then create C=>A, it just replaces it by C=>B. No need to chain because it makes the recounting more tricky, and doesn't add any functionality. It's strictly equivalent to have all storages point to the same end storage.